core/src/main/scala/org/apache/spark/sql/execution/ddl.scala

```diff
@@ -407,14 +407,14 @@ class ShowSnappyTablesCommand(session: SnappySession, schemaOpt: Option[String],
   override val output: Seq[Attribute] = {
     if (hiveCompatible) AttributeReference("name", StringType, nullable = false)() :: Nil
     else {
-      AttributeReference("schemaName", StringType, nullable = false)() ::
+      AttributeReference("database", StringType, nullable = false)() ::
           AttributeReference("tableName", StringType, nullable = false)() ::
           AttributeReference("isTemporary", BooleanType, nullable = false)() :: Nil
     }
   }
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    if (!hiveCompatible) return super.run(sparkSession)
+    if (!hiveCompatible) return super.run(sparkSession).map(r => Row(r.getString(0).toLowerCase, r.getString(1).toLowerCase, r.getBoolean(2)))
```

Comment:
The change is specific to make Spotfire work. 
Do we know whether we have to convert to lower case even when hive compatibility is turned on?

Reply:
Now changed Catalog listDatabases/listTables itself to return the result as received from hive client (which is lower-case) so that it is Spark compatible. Other code changes to deal with this otherwise have been added.